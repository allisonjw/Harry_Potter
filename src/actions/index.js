export const addCharacters = characters => ({
    type: 'ADD_CHARACTERS',
    characters
});

export const addHouses = houses => ({
    type: 'ADD_HOUSES',
    houses
});

export const addSpells = spells => ({
    type: 'ADD_SPELLS',
    spells
});

export const isLoading = isLoading => ({
    type: 'SET_LOADING',
    isLoading
});

export const addHouseCharacters = filteredStudents => ({
    type: 'SET_CHARACTERS',
    filteredStudents
});

export const addFavorite = (id) => ({
    type: 'ADD_FAVORITE',
    id
});
  
export const deleteFavorite = (id) => ({
    type: 'DELETE_FAVORITE',
    id
});