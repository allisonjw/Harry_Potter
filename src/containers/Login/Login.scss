@import url("https://fonts.googleapis.com/css?family=Henny+Penny|ZCOOL+XiaoWei&display=swap");

// @font-face {
//     font-family: 'Harry Potter Regular';
//     font-style: normal;
//     font-weight: normal;
//     src: local('Harry Potter Regular'), url('HARRP___.woff') format('woff');
//     }

.login_main {
    background-image: url(https://images.ctfassets.net/usf1vwtuqyxm/7EsDSrJhr0pNwUNK8lrUwY/f8d7e5101de6dda8f46c0f94673087a9/HP-Hogwarts-39PHOTOBU17074_PHUP_HP.jpg);
    align-items: center;
    background-size: cover;
    display: flex;
    flex-direction: column;
    height: 100vh;
}

#blockColorblindContent {
    display: none;
}

h1 {
    font-family: 'Henny Penny', cursive;
    // font-family: 'Harry Potter Regular';
    font-size: 4em;
    color: white;
}

.login_witchcraft {
    background-color: #aca9a3;
    background-image: url("https://www.transparenttextures.com/patterns/brick-wall-dark.png");
    // font-family: 'Harry Potter Regular';
    font-family: 'Henny Penny', cursive;
    font-size: 3em;
    text-transform: uppercase;
    color: #252627;
    margin-top: 25%;
}

.submit-btn {
    border-radius: 3px;
    font-family: 'Henny Penny', cursive;
    font-size: 2em;
    height: 60px;
    margin-bottom: 10%;
}