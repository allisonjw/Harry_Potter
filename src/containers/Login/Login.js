import React, { Component } from 'react';
import './Login.scss';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { addUser } from '../../actions';
import { bindActionCreators } from 'redux';
// import { PropTypes } from 'prop-types';

export class Login extends Component{
    constructor() {
        super()
        this.state = {
            name: '',
            wizardType: '',
            house: ''
        }
    };

    handleName = (e) => {
        this.setState({name: e.target.value})
    };
    
    handleWizardType = (e) => {
      this.setState({[e.target.name]: e.target.value})
    };
    
    handleUser = () => {
        const { addUser } = this.props
        const user = {
            name: this.state.name,
            wizardType: this.state.wizardType,
            house: this.state.house
        };
        addUser(user)
        this.clearInputs();
    };

    clearInputs = () => {
        this.setState = ({
            name: '',
            wizardType: '',
            house: ''
        });
    };


    render() {
        const { user } = this.props;
        const house = user === undefined ? '' : user.house;
        let submitButton;

        if(this.state.name !== '' && this.wizardType !== '') {
          submitButton = (
             <button className='submit_button' 
                type='button' 
                onClick={this.handleUser}>Enter</button>
          )
        };
        
        return (
            <section className="login_main">
              <h1>Harry Potter</h1>
              <h2 className='login_witchcraft'>Witchcraft and Wizardry</h2>
            <form className='login_form'>
                <p className='login_msg'>Enter your name and select Witch or Wizard</p>
                    <input className='name_input' 
                        type='text' 
                        placeholder='Enter Name' 
                        onChange={this.handleName}/>
                <h2 className="wizardType_label">Witch or Wizard</h2>
                <div className='wizardType_buttons'>
                    <button className='wizardType_button'
                        type='button' 
                        name='wizardType' 
                        value='Witch'
                        onClick={this.handleWizardType}>Witch
                    </button>
                    <button className='wizardType_button'
                        type='button' 
                        name='wizardType' 
                        value='Wizard' 
                        onClick={this.handleWizardType}>Wizard
                    </button>
                </div>
                    <Link to='/houses' >
                        {submitButton}
                    </Link>
            </form>      
            </section>
        )
    }
};

export const mapStateToProps = ({ user }) => ({
    user
});

export const mapDispatchToProps = dispatch => {
    return bindActionCreators({ 
        addUser: user => dispatch( addUser(user) ) 
    }, dispatch)
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);