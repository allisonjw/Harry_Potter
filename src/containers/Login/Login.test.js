import React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import { addUser } from '../../actions/index';
import { Login, mapStateToProps, mapDispatchToProps } from './Login';

describe('Login', () => {
    let wrapper;
    const mockAddUser = jest.fn();
    const mockHouse = ''
    const mockUser = {
      name: 'name',
      wizardType: 'wizard',
      house: ''
    }
    
    beforeEach(() => {
      wrapper = shallow (<Login 
        addUser={mockAddUser} 
        house={mockHouse}
        user={mockUser}
      />);
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should update state when handleName is called',() => {
      const mockEvent = {
        target: {
          value: 'name'
        }
      }
      const expected = 'name'
      wrapper.instance().handleName(mockEvent)
      expect(wrapper.state('name')).toEqual(expected);
    });

    it('should update state when handleWizardType is called',() => {
      const mockEvent = {
        target: {
          name: 'wizardType',
          value: 'Witch'
        }
      };

      const expected = 'Witch'
      wrapper.instance().handleWizardType(mockEvent)
      expect(wrapper.state('wizardType')).toEqual(expected);
    });

    it('should create a new user when handleUser is called', () => {
      const mockUser = {
        name: '',
        wizardType: '',
        house: ''
      };

      wrapper.instance().handleUser();
      expect(mockAddUser).toHaveBeenCalledWith(mockUser)
    });

    it('should update state when clearInputs is called', () => {
      const mockState = {
        name: 'name',
        wizardType: 'witch',
        house: ''
      };

      const expected = ''
      wrapper.instance().clearInputs()
      expect(wrapper.state('name')).toEqual(expected);
    });

    describe('mapStateToProps', () => {
        it('should return an user object', () => {
            const mockUser =  { name: 'Allison', wizardType: 'witch', house: '' }

            const mockState = {
                user: mockUser
              };
        
              const expected = {
                  user: mockUser,
              };
        
              const mappedProps = mapStateToProps(mockState);
              expect(mappedProps).toEqual(expected);
        })
    });

    describe('mapDispatchToProps', () => {
        it('should call dispatch with the correct action when addUser is called', () => {
        const mockDispatch = jest.fn();
        const mockDispatchUser = { name: 'Allison', wizardType: 'Witch', house: '' };
        const dispatchedAction = addUser(mockDispatchUser);
        const mappedProps = mapDispatchToProps(mockDispatch);
        mappedProps.addUser(mockDispatchUser);
        expect(mockDispatch).toHaveBeenCalledWith(dispatchedAction);
        });
    });
});