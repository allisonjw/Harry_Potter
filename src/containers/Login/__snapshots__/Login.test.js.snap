// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `
<section
  className="login_main"
>
  <h1>
    Harry Potter
  </h1>
  <h2
    className="login_witchcraft"
  >
    Witchcraft and Wizardry
  </h2>
  <form
    className="login_form"
  >
    <p
      className="login_msg"
    >
      Enter your name and select Witch or Wizard
    </p>
    <input
      className="name_input"
      onChange={[Function]}
      placeholder="Enter Name"
      type="text"
    />
    <h2
      className="wizardType_label"
    >
      Witch or Wizard
    </h2>
    <div
      className="wizardType_buttons"
    >
      <button
        className="wizardType_button"
        name="wizardType"
        onClick={[Function]}
        type="button"
        value="Witch"
      >
        Witch
      </button>
      <button
        className="wizardType_button"
        name="wizardType"
        onClick={[Function]}
        type="button"
        value="Wizard"
      >
        Wizard
      </button>
    </div>
    <Link
      to="/houses"
    />
  </form>
</section>
`;
